source/exe/win32/service_base.cc

```diff
@@ -50,8 +50,11 @@ bool ServiceBase::TryRunAsService(ServiceBase& service) {
     if (last_error == ERROR_FAILED_SERVICE_CONTROLLER_CONNECT) {
       return false;
     } else {
-      PANIC(
-          fmt::format("Could not dispatch Envoy to start as a service with error {}", last_error));
+      std::string error_msg{
+          fmt::format("Could not dispatch Envoy to start as a service with error {}", last_error)};
+      service_static->windows_event_logger_.log(::spdlog::source_loc{__FILE__, __LINE__, __func__},
+                                                spdlog::level::err, error_msg);
```

Comment:
It seems like we could define some macro here even if not the normal ones for this so we avoid the repetition?

Reply:
General aversion towards C style macro's but I guess they are needed here to preserve the source location. added in b82744e1e064c95fa55788297120929d835372a8